Packages/nadena.dev.modular-avatar/Editor/ObjectRemoverProcessor.cs

```diff
@@ -0,0 +1,30 @@
+﻿using UnityEditor;
+using UnityEngine;
+
+namespace nadena.dev.modular_avatar.core.editor
+{
+    public static class ObjectRemoverProcessor
+    {
+        internal static void OnPreprocessAvatar(GameObject avatarGameObject)
+        {
+            var toRemove = avatarGameObject.transform.GetComponentsInChildren<ModularAvatarObjectRemover>(true);
+
+            foreach (var remove in toRemove)
+            {
+                Object.DestroyImmediate(remove.gameObject);
+            }
+        }
+
+
+        [MenuItem("GameObject/ModularAvatar/Restore Removed", false, 49)]
```

Comment:
There should probably be a validate function here to disable this when there are no hidden/removed objects.

Reply:
no longer needed